plotly/tests/test_core/test_tools/test_trace_factory.py

```diff
@@ -100,3 +100,180 @@ def test_more_kwargs(self):
             'type': 'scatter',
             'mode': 'lines', }
         self.assertEqual(quiver, expected_quiver)
+
+
+class TestOHLC(TestCase):
+
+    def test_unequal_o_h_l_c_length(self):
+
+        # check: PlotlyError if op, hi, lo, cl are not the same length for
+        # TraceFactory.ohlc_increase and TraceFactory.ohlc_decrease
+
+        kwargs = {'op': [1], 'hi': [1, 3], 'lo': [1, 2], 'cl': [1, 2]}
+        self.assertRaises(PlotlyError, tls.TraceFactory.create_ohlc_increase,
+                          **kwargs)
+
+        kwargs = {'op': [1, 2], 'hi': [1, 2, 3], 'lo': [1, 2], 'cl': [1, 2]}
+        self.assertRaises(PlotlyError, tls.TraceFactory.create_ohlc_increase,
+                          **kwargs)
+
+        kwargs = {'op': [1], 'hi': [1, 3], 'lo': [1, 2], 'cl': [2]}
+        self.assertRaises(PlotlyError, tls.TraceFactory.create_ohlc_decrease,
+                          **kwargs)
+
+        kwargs = {'op': [1, 2], 'hi': [1, 2], 'lo': [1, 2, .5], 'cl': [1, 2]}
+        self.assertRaises(PlotlyError, tls.TraceFactory.create_ohlc_decrease,
+                          **kwargs)
+
+    def test_hi_highest_value(self):
+
+        # check: PlotlyError if the "hi" value is less than the corresponding
+        # op, lo, or cl value because if the "high" value is not the highest
+        # (or equal) then the data may have been entered incorrectly.
+
+        # create_ohlc_increase
+        kwargs = {'op': [2, 3], 'hi': [4, 2], 'lo': [1, 1], 'cl': [1, 2]}
+        self.assertRaises(PlotlyError, tls.TraceFactory.create_ohlc_increase,
+                          **kwargs)
+
+        # create_ohlc_decrease
+        kwargs = {'op': [2, 3], 'hi': [4, 3], 'lo': [1, 1], 'cl': [1, 6]}
+        self.assertRaises(PlotlyError, tls.TraceFactory.create_ohlc_decrease,
+                          **kwargs)
```

Comment:
should we check any of these messages? if they're supposed to be specific about the error, we might want to lock it down.